test/e2e/lib/components/shopping-cart-widget-component.js

```diff
@@ -43,4 +43,31 @@ export default class ShoppingCartWidgetComponent extends AsyncBaseContainer {
 			}
 		}
 	}
+
+	async removeDomainRegistraion( domain ) {
+		return this.remove( 'domain_reg', domain );
+	}
+
+	async removeDomainMapping( domain ) {
+		return this.remove( 'domain_map', domain );
+	}
+
+	async remove( type, name ) {
+		const cartBadgeLocator = by.css( '.cart__count-badge' );
+
+		const present = await driverHelper.isElementLocated( this.driver, cartBadgeLocator );
+		if ( present ) {
+			await this.open();
+			await driverHelper.clickWhenClickable(
+				this.driver,
+				by.xpath(
+					// Find an element X with class=.cart-item
+					//    that contains an element with data-e2e-product-slug=`type`
+					//    and a sibling with class="product-domain" and text=`name`
+					// and then select an element inside X that matches class=cart__remove-item
+					`//*[@class="cart-item"][.//*[@data-e2e-product-slug="${ type }"]/following-sibling::*[@class="product-domain"][text()="${ name }"]]//*[@class="cart__remove-item"]`
```

Comment:
Woah, that's an impressive selector right there! 😄 FYI, there's also the [`RelativeBy`](https://github.com/SeleniumHQ/selenium/blob/trunk/javascript/node/selenium-webdriver/lib/by.js#L270-L374) location strategies that might be useful in such cases.